src/unix/kqueue.c

```diff
@@ -285,13 +288,13 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
       if (reset_timeout != 0) {
         timeout = user_timeout;
         reset_timeout = 0;
-      } else if (nfds == 0) {
-        /* Reached the user timeout value. */
-        assert(timeout != -1);
-        return;
```

Comment:
I think you still want this return, so that you don't get stuck here if the user removes all event sources during the loop callbacks?

Reply:
Ah, I see this is what you meant by updating the timestamp. I guess that makes sense then. And seems to DRY the code even more.

Reply:
That does sound correct, but should we also make sure to run `uv__update_time()` before returning?